app/src/main/java/com/github/onlynotesswent/utils/NotesToFlashcard.kt

```diff
@@ -74,25 +111,219 @@ class NotesToFlashcard(
           val flashcardObject = element.asJsonObject
           val question = flashcardObject.get("question").asString
           val answer = flashcardObject.get("answer").asString
+          val latexFormula = flashcardObject.get("latexFormula")?.asString ?: ""
+          val fakeBacks =
+              flashcardObject.getAsJsonArray("fakeBacks")?.map { it.asString } ?: emptyList()
 
           val flashcard =
               Flashcard(
                   id = flashcardViewModel.getNewUid(),
                   front = question,
                   back = answer,
+                  latexFormula = latexFormula,
+                  hasImage = false,
+                  fakeBacks = fakeBacks,
                   lastReviewed = null,
                   userId = note.userId,
-                  folderId = note.folderId,
+                  folderId = folderId,
                   noteId = note.id)
           flashcards.add(flashcard)
           flashcardViewModel.addFlashcard(flashcard)
         }
       }
     } catch (e: Exception) {
       Log.e(TAG, "Error parsing flashcards from JSON", e)
+      println("Error parsing flashcards from JSON $e")
       onFailure(e)
       return
     }
-    onSuccess(flashcards)
+    val deck =
+        Deck(
+            id = deckViewModel.getNewUid(),
+            name = note.title,
+            userId = note.userId,
+            folderId = folderId,
+            visibility = note.visibility,
+            lastModified = Timestamp.now(),
+            flashcardIds = flashcards.map { it.id })
+    deckViewModel.updateDeck(deck)
+    onSuccess(deck)
+  }
+
+  /**
+   * Converts a folder and its subfolders into decks of flashcards.
+   *
+   * This method starts the conversion process for the selected folder and ensures all its
+   * subfolders are processed recursively. Each note in the folder is converted into a deck of
+   * flashcards, and combined decks are created for a folder if it contains multiple notes.
+   *
+   * @param onSuccess Callback invoked when the conversion is successful with the final deck.
+   * @param onFailure Callback invoked when the conversion process fails with an exception.
+   */
+  fun convertFolderToDecks(onSuccess: (Deck) -> Unit, onFailure: (Exception) -> Unit) {
```

Comment:
while you correct the other comment with the boolean array and if you feel like it, you could pass a lambda to the ui which gets called for each note / folder processed (and would for example increment a counter in the ui). This way there could be a fun loading screen showing actual progress, eg: `Loading...  23 notes converted in 4 folders`.

just a thought

Reply:
I pass in a onProgress and I pass it the number of notes, folder that have been processed. I also pass in any exception that has occurred during the conversion. I track the note and folder with AtomicInteger so that they are correctly updated in the different coroutines.